include/boost/math/differentiation/autodiff.hpp

```diff
@@ -567,13 +567,36 @@ template<typename RealType, size_t Order, size_t... Orders>
 using autodiff_fvar = typename detail::nest_fvar<RealType,Order,Orders...>::type;
 
 template<typename RealType, size_t Order, size_t... Orders>
-autodiff_fvar<RealType,Order,Orders...> make_fvar(const RealType& ca)
-{
+autodiff_fvar<RealType,Order,Orders...> make_fvar(const RealType& ca) {
   return autodiff_fvar<RealType,Order,Orders...>(ca, true);
 }
 
 namespace detail {
 
+template <size_t>
+struct zero : std::integral_constant<size_t,0> {};
+
+template<typename RealType, size_t Order, size_t... Is>
+auto make_fvar_for_tuple(mp11::index_sequence<Is...>, const RealType& ca) -> decltype(make_fvar<RealType,zero<Is>::value...,Order>(ca)) {
+  return make_fvar<RealType,zero<Is>::value...,Order>(ca);
+}
+
+template<typename RealType, size_t... Orders, size_t... Is, typename... RealTypes>
+auto make_ftuple(mp11::index_sequence<Is...>, const RealTypes&... ca) -> decltype(std::make_tuple(make_fvar_for_tuple<RealType,Orders>(mp11::make_index_sequence<Is>{},ca)...)) {
```

Comment:
I think you can omit the ->type as of c++14. Probably makes sense to where it's decl type of the complete return expression

Reply:
You’re correct; this is only needed for C++11. It’s rather ugly, though.

Reply:
I somehow forgot that c++11 support was a thing. I guess we need this afterall.